r/src/array_to_vector.cpp

```diff
@@ -919,8 +920,9 @@ class Converter_Decimal : public Converter {
 
   Status Ingest_some_nulls(SEXP data, const std::shared_ptr<arrow::Array>& array,
                            R_xlen_t start, R_xlen_t n, size_t chunk_index) const {
+    using DecimalArray = typename TypeTraits<Type>::ArrayType;
     auto p_data = REAL(data) + start;
-    const auto& decimals_arr = checked_cast<const arrow::Decimal128Array&>(*array);
+    const auto& decimals_arr = checked_cast<const DecimalArray&>(*array);
 
     auto ingest_one = [&](R_xlen_t i) {
       p_data[i] = std::stod(decimals_arr.FormatValue(i).c_str());
```

Comment:
Unrelated, but this looks quite inefficient (it roundtrips through a string representation instead of creating a `double` directly). Can you open a JIRA for this?

Reply:
I've created https://issues.apache.org/jira/browse/ARROW-15167

Reply:
Cool. Thanks. I was just about to create it.